src/common/utils.h

```diff
@@ -31,70 +31,55 @@ template<typename xpu>
 void CastStorageDispatch(const OpContext& ctx, const NDArray& input, const NDArray& output);
 
 /*
- * \brief Get the corresponding tensor blobs from default storage NDArrays.
- *        If any NDArray is of non-default storage, it is casted to default storage and
- *        the temporary NDArrays are stored in `temps`. When storage_fallback is false,
- *        and `MXNET_EXEC_STORAGE_FALLBACK` == 0, storage fallback is disallowed.
- * \return true if any input is casted
+ * \brief get the corresponding tensor blobs from default storage NDArrays.
+ *        If any NDArray is of non-default storage, it will be added to `temp_src`
+ * \return true if any input storage needs to be casted
  */
-template <typename xpu>
-inline bool GetDefaultBlobs(const std::vector<NDArray>& nds,
+inline bool GetDefaultBlobs(const std::vector<NDArray>& src,
                             std::vector<TBlob> *blobs,
-                            std::vector<NDArray> *temps,
-                            const OpContext& ctx,
-                            bool storage_fallback = false) {
-  bool casted = false;
-  if (storage_fallback == false) {
-    storage_fallback = dmlc::GetEnv("MXNET_EXEC_STORAGE_FALLBACK", true);
-  }
-  for (auto& nd : nds) {
+                            std::vector<NDArray> *temp_src,
+                            std::vector<NDArray> *temp_dst) {
+  bool require_cast = false;
+  size_t size = src.size();
```

Comment:
It is not as bad as "bad practice", but not recommended due to the readability and maintainability issues. For example, if some day, a user inserted more ndarrays to `src` between lines 43 and 44, the `for loop` might be incomplete.